src/dispatching/dispatchers/filter.rs

```diff
@@ -0,0 +1,382 @@
+//! A dispatcher based on filters.
+
+use futures::StreamExt;
+
+use crate::{
+    dispatching::{filters::Filter, ErrorHandler, Handler, Updater},
+    types::{
+        CallbackQuery, ChosenInlineResult, InlineQuery, Message, Update,
+        UpdateKind,
+    },
+};
+
+type FilterWithHandler<'a, T, E> =
+    (Box<dyn Filter<T> + 'a>, Box<dyn Handler<T, E> + 'a>);
+type FiltersWithHandlers<'a, T, E> = Vec<FilterWithHandler<'a, T, E>>;
+
+/// A dispatcher based on filters.
+///
+/// Filters and handlers are executed in order of registering. The pseudocode
+/// looks like this:
+///
+/// ```no
+/// for pair in handlers_and_filters {
+///     if pair.filter.test(update) {
+///         pair.handle(update);
+///         return;
+///     }
+/// }
+///
+/// log("unhandeled update: " + update);
+/// ```
+///
+/// ## Examples
+///
+/// Simplest example:
+/// ```no_run
+/// # async fn run() {
+/// use std::convert::Infallible;
+///
+/// use teloxide::{
+///     dispatching::{
+///         dispatchers::filter::FilterDispatcher, updaters::polling_basic,
+///     },
+///     types::Message,
+///     Bot,
+/// };
+///
+/// async fn handle_edited_message(mes: Message) -> Result<(), Infallible> {
+///     println!("Edited message: {:?}", mes);
+///     Ok(())
+/// }
+///
+/// let bot = Bot::new("TOKEN");
+///
+/// // create dispatching which handlers can't fail
+/// // with error policy that just ignores all errors (that can't ever happen)
+/// let mut dp = FilterDispatcher::<Infallible, _>::new(|_| async {})
+///     // Add 'handler' that will handle all messages sent to the bot
+///     .message_handler(true, |mes: Message| async move {
```

Comment:
I think here need comment why first argument is `true`

Reply:
> // Add 'handler' that will handle all messages sent to the bot

Reply:
Probably we need to add `... and this is why the first argument is true`.

Reply:
I think the code must be self-documented, so we probably need something like `Filte::ALL`.

Reply:
I'm ok, with that, but we need a different naming (and this should be in another PR):
1) we have no struct called `Filter`
2) `ALL` sounds like combinator and we also have `all!` macro that combines any amount of filters.
3) It's confusing: does `ALL` means "pass anything" or "doesn't pass anything"?

After thinking for a while I come up with this
1) current `true`/`false` approach is the best because of the hard naming. Also it can be well documented.
2) another approach can be `filters::Bool::{True, False}` or just `filters::{True, False}` (later a bit conflicts with `types::{True, False}`) maybe it would be less confusing.
3) `struct Always(bool)` meeeeh, but :shrug:

Reply:
> It's confusing: does ALL means "pass anything" or "doesn't pass anything"?

Pass everything.

> current true/false approach is the best because of the hard naming. Also it can be well documented.

The current `true`/`false` approach is the worst one because it's not self-documenting, after a month of writing a project I must look at the signature and conversions to understand what actually `true`/`false` means.

Reply:
> Pass everything.  

I know, but it's just not intuitive